packages-exp/auth-compat-exp/src/auth.ts

```diff
@@ -60,53 +59,69 @@ export class Auth
       sdkClientVersion: impl._getClientVersion(_getClientPlatform())
     };
 
-    super(app, config, User);
-    this.app = app;
+    this.auth = new impl.AuthImpl(app, config);
```

Comment:
The modular auth instance should be retrieved from the component framework and passed into the compat auth constructor instead of being created manually. Otherwise, auth-exp and auth-compat won't share the same state.

In compat auth factory:
```
        const app = container.getProvider('app').getImmediate();
        const auth = container.getProvider('auth-exp').getImmediate();
        return new Auth(app, auth);
```

It probably also means you need to register `auth-exp` with the component framework in '@firebase/auth-exp/internal'. You can move the registration code to a separate file, e.g. `register.ts` and make both `@firebase/auth-exp` and `@firebase/auth-exp/internal` import it.

Reply:
I tried this and I kept getting the error that it couldn't find the provider:

```
Uncaught Error: Service auth-exp is not available
    at Provider.getImmediate (provider.ts:101)
```

I've pushed up this version so you can take a look. Not sure what's going wrong

Reply:
Ah I see. It's the script issue with compat packages we chatted about before. Will report back once I get the compat script work.

Reply:
Would you be willing to let me merge this with a TODO? Many of the remaining parity issues on my plate deal with the compat layer.

Reply:
https://github.com/firebase/firebase-js-sdk/pull/3916
It should work with this PR. I tested it locally. 

Sure, if this PR doesn't solve the problem completely, we can merge it as this with a TODO.